modules/nextflow/src/main/groovy/nextflow/splitter/CsvSplitter.groovy

```diff
@@ -107,6 +124,7 @@ class CsvSplitter extends AbstractTextSplitter {
         result.header = [ Boolean, List ]
         result.quote = String
         result.skip = Integer
+        result.types = [ Boolean, List ]
```

Comment:
@mirpedrol related to the other thread, I totally forgot about this -- the `types` option should also be allowed to be a map, so that you can assign types by column name. It should be allowed only if a header row is expected.

Reply:
Not sure about this. IMO it will be more verbose. The explicit list of types sounds better

Reply:
We should support both. Again, see `pandas.read_csv` for a good example of this